### PR TITLE
Proposal: Use SVGR components for UI icons

### PR DESCRIPTION
**Pull request by CPatchane**
This PR brings SVG components using [svgr](https://github.com/smooth-code/svgr) to the `<Icon />` instead of using the sprite `<IconSprite/>`. 

Results/Preview available here: https://cpatchane.github.io/cozy-ui/react/#icon

### Why ?
For me, the application should not worry about adding a sprite to use a component from a UI framework. Using this way, the application just have to use `<Icon />` from cozy-ui and the only used icons will be added to the app code.

### Breaking changes ?

I don't think so. The `<Icon />` will use these components by default when you will pass a string which do not start by `#` as the `icon` props. If it find a component matching this name, it will use the related SVG component. If not it will use the id as before (#name).
If we really want to force using the sprite from the application, a props `forceSprite` can be passed to force using sprite in all cases.

### Anything else ?

This PR also remove all these icons with colour from `<Icon />` usage (not from cozy-ui): `delete-grey08`, `share-grey08`, `upload-grey08`, `dots-white`, `cross-white`, `spinner-red`, `spinner-blue`, `spinner-grey`, `spinner-white`.
They shouldn't be usedby `<Icon />` since there is a colour option and you can't use this option with those.

Fix https://github.com/cozy/cozy-ui/issues/705 
Fix https://github.com/cozy/cozy-ui/issues/674

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,8 +35,9 @@
     "test:jest": "env BABEL_ENV=transpilation jest --verbose --coverage",
     "transpile": "env BABEL_ENV=transpilation babel react/ --out-dir transpiled/react --copy-files --ignore '**/test/**,**/__mocks__/**'",
     "sprite": "scripts/make-icon-sprite.sh transpiled/react/Icon/icons-sprite.js; cp transpiled/react/Icon/icons-sprite.js react/Icon/icons-sprite.js",
+    "svgComp": "svgr --icon --no-dimensions --prettier-config .prettierrc.json -d 'react/_icons' 'assets/icons/ui'",
     "copy-files": "scripts/copy-files.sh",
-    "release": "yarn transpile; yarn sprite; yarn copy-files",
+    "release": "npm-run-all --sequential transpile sprite copy-files",
```
Comment (ptbrowne) [inline package.json]: What is the advantage of doing this ?

Comment (ptbrowne) [reply package.json]: npm-run-all exposes the run-s shortcut.

File: react/Icon/index.jsx [hunk 2]
```diff
@@ -2,6 +2,8 @@ import React from 'react'
 import PropTypes from 'prop-types'
 import styles from './styles.styl'
 import cx from 'classnames'
+import toPascalCase from 'to-pascal-case'
+import * as iconsIndex from './icons'
```
Comment (ptbrowne) [inline react/Icon/index.jsx]: >  and the only used icons will be added to the app code.

With this, all icons are added.

**Comment by ptbrowne** [comment]
LGTM, thanks Cedric !

**Comment by Crash--** [comment]
Thank you @CPatchane for this work! I like the idea of not having to include any sprite in our app. 

To resume: 

- We remove the need of using a sprite by using an IconComponent. But if we use 10 times the same icon, we'll have ten times the icon into the DOM. 
- Having an icon component means that we can't override it anymore. Is it something we really want for UI's icons @edas ? 

Some questions / remarks :
- It seems that it will be retro compatible. So we'll be able to use Icon sprite and icon component at the same time? 
- Can we edit the doc to explain how to add a new icon to cozy ui with this new system?

**Comment by CPatchane** [comment]
> It seems that it will be retro compatible. So we'll be able to use Icon sprite and icon component at the same time?

Yes, you still can import a sprite in your application and use the it via the `forceSprite` props here

**Comment by edas** [comment]
>     * Having an icon component means that we can't override it anymore. Is it something we really want for UI's icons @edas ?

No we don't
What we are trying to achieve, is a way to be able to override all icons and colors. We probably should put in a sprite or external file all SVG which are now in base64, not the opposite.

Icons that respect a theme (black & white themable by CSS) may be different. The question is : do we really want them to be inserted multiple times in the DOM if we use them more than once ?

**Comment by CPatchane** [comment]
@edas So this PR is not a solution that we want about SVG handling here. It may be better to see what we can do with our sprite tools instead. Thanks!

**Pull request closed without merging**